src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -253,11 +253,11 @@ defaultEnqueuePolicyEdgeBehindNat = go
         -- not forwarded
       ]
     go MsgRequestBlockHeaders = [
-        EnqueueAll NodeRelay (MaxAhead 1) PHigh
+        EnqueueAll NodeRelay (MaxAhead 0) PHigh
       ]
-    go (MsgRequestBlock _) = [
+    go (MsgRequestBlocks _) = [
         -- Edge nodes can only talk to relay nodes
-        EnqueueOne [NodeRelay] UnlimitedMaxAhead PHigh
+        EnqueueOne [NodeRelay] (MaxAhead 0) PHigh
```

Comment:
Because this is the highest priority message that edge nodes send. Edge nodes do not send block header announcements, so we do not expect to find block header announcements in the queue ahead of us. Whereas for core and relay nodes, they do send block header announcements, and those take the maxahead 0 position.